src/app/static/src/app/store/metadata/mutations.ts

```diff
@@ -1,23 +1,38 @@
-import {Mutation, MutationTree} from 'vuex';
+import {MutationTree} from 'vuex';
 import {MetadataState} from "./metadata";
-import {PlottingMetadataResponse, Error} from "../../generated";
+import {PlottingMetadataResponse, Error, AdrMetadataResponse} from "../../generated";
 import {PayloadWithType} from "../../types";
 
-type MetadataMutation = Mutation<MetadataState>
-
-export interface MetadataMutations {
-    PlottingMetadataFetched: MetadataMutation
-    PlottingMetadataError: MetadataMutation
+export enum MetadataMutations {
+    PlottingMetadataFetched= "PlottingMetadataFetched",
+    PlottingMetadataError = "PlottingMetadataError",
+    AdrUploadMetadataFetched = "AdrUploadMetadataFetched",
+    AdrUploadMetadataError = "AdrUploadMetadataError"
 }
 
-export const mutations: MutationTree<MetadataState> & MetadataMutations = {
+export const mutations: MutationTree<MetadataState> = {
 
-    PlottingMetadataError(state: MetadataState, action: PayloadWithType<Error>) {
+    [MetadataMutations.PlottingMetadataError](state: MetadataState, action: PayloadWithType<Error>) {
         state.plottingMetadataError = action.payload;
     },
 
-    PlottingMetadataFetched(state: MetadataState, action: PayloadWithType<PlottingMetadataResponse>) {
+    [MetadataMutations.PlottingMetadataFetched](state: MetadataState, action: PayloadWithType<PlottingMetadataResponse>) {
         state.plottingMetadata = action.payload;
         state.plottingMetadataError = null;
+    },
+
+    [MetadataMutations.AdrUploadMetadataError](state: MetadataState, action: PayloadWithType<Error>) {
+        state.adrUploadMetadataError = action.payload;
+    },
+
+    [MetadataMutations.AdrUploadMetadataFetched](state: MetadataState, action: PayloadWithType<AdrMetadataResponse>) {
+        const adrMetadataTypeExists = state.adrUploadMetadata
+            .findIndex(adrMeta => adrMeta.type === action.payload.type)
+
+        adrMetadataTypeExists > -1 ?
+            state.adrUploadMetadata[adrMetadataTypeExists].description = action.payload.description
+            : state.adrUploadMetadata.push({type: action.payload.type, description: action.payload.description})
+
```

Comment:
Would this be a bit less of a faff if `adrUploadMetadata` was a dictionary keyed on type rather than an array?